### ISSUE TITLE
GoCD incoming slack webhooks should ask for approval or reject option

### ISSUE THREAD

**Issue by zerolaser** [ashwanthkumar/gocd-slack-build-notifier#74]
Is it possible for the Gocd Slack notification to ask for approve or reject button.

lets say a pipeline has two stages. soon after successful run of stage1 it should send a notification to slack channel. Do you approve or reject it ? 
when approval button is clicked it should run stage2

**Comment by ashwanthkumar** [comment]
@zerolaser I really like the approach. But AFAIK to build action buttons and make it work, slack needs to be able to access an action url to post the status. We generally don't want to expose our CI / CD servers to the public internet.

**Comment by zerolaser** [comment]
@ashwanthkumar yeah. its a quite risk exposing it. what if you add an authentication mechanism to do so ? 
will it be a risk of exposing GoCD server's user credentials to slack (incoming webhook)?

most of the cases slack and gocd will be on ldap authentication, get the slack userid and match it with GoCD user and also check if the user has permission to trigger the pipeline? 

Also I am looking forward for a web interface on GoCD to configure slack.
